java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/BigIntegerInstantiation.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025 Benoit Lacelle - SOLVEN
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.solven.cleanthat.engine.java.refactorer.mutators;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.FieldAccessExpr;
+import com.github.javaparser.ast.expr.LiteralStringValueExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.ObjectCreationExpr;
+import com.google.common.collect.ImmutableSet;
+
+import eu.solven.cleanthat.engine.java.IJdkVersionConstants;
+import eu.solven.cleanthat.engine.java.refactorer.AJavaparserExprMutator;
+import eu.solven.cleanthat.engine.java.refactorer.NodeAndSymbolSolver;
+import eu.solven.cleanthat.engine.java.refactorer.meta.IMutatorDescriber;
+
+/**
+ * Turns `new BigInteger("1")` into `BigInteger.ONE`
+ *
+ * @author Balazs Glatz
+ */
+public class BigIntegerInstantiation extends AJavaparserExprMutator implements IMutatorDescriber {
+
+	private static final String NOTHING = "";
+
+	private static final String NUMERIC_LITERAL_SUFFIX_REGEX = "(?<=\\d)(\\.0?)?([ldfLDF])?$";
```

Comment:
Please document regex. One should not have to parse the reges to know what's it achieving.

Reply:
Supposing what's doing this regex, I feel it does not manage cases like: `double d = 1.0e0;`. I suggest either finding another way (doing a plain `Double.parse` maybe?), or relying on a solid regex, with a solid reference (e.g., some stackoverflow with many votes).